### PR TITLE
Document EncoderConfig

### PR DESCRIPTION
**Pull request by vibhoothiiaanand**
Add documentation of the encoderConfig for both impl and  public fields
Ref #1078

### CODE REVIEW

File: src/api.rs [hunk 1]
```diff
@@ -83,6 +83,8 @@ pub struct EncoderConfig {
   pub train_rdo: bool,
 }
 
+/// Default preset for EncoderConfig,it is a balance between quality and speed.
```
Comment (lu-zero) [inline src/api.rs]:
```suggestion
/// Default preset for EncoderConfig: it is a balance between quality and speed.
```

File: src/api.rs [hunk 2]
```diff
@@ -83,6 +83,8 @@ pub struct EncoderConfig {
...
+/// User can use with_speed_preset to have a default structure with a specific speed level.
```
Comment (lu-zero) [inline src/api.rs]:
```suggestion
/// See [EncoderConfig::with_speed_preset].
```

File: src/api.rs [hunk 3]
```diff
@@ -91,6 +93,9 @@ impl Default for EncoderConfig {
 }
 
 impl EncoderConfig {
+
+  /// This is a preset which provides default settings according to a speed value in the specific range 0-10,
```
Comment (lu-zero) [inline src/api.rs]: You should reword it so it points to the SpeedSetting documentation maybe.

**Comment by tdaede** [comment]
@lu-zero are your changes addressed now?

**Pull request merged**